library/src/main/java/com/schibsted/spain/barista/BaristaRecyclerViewActions.java

```diff
@@ -24,21 +26,21 @@ public static void clickRecyclerViewItem(@IdRes int recyclerViewId, int... posit
   }
 
   private static void performClick(@IdRes int recyclerViewId, int position) {
-    onView(withId(recyclerViewId)).perform(
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(
         RecyclerViewActions.actionOnItemAtPosition(position, clickUsingPerformClick()));
   }
 
   public static void scrollTo(int recyclerViewId, int position) {
-    onView(withId(recyclerViewId)).perform(scrollToPosition(position));
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(scrollToPosition(position));
   }
 
   public static void clickRecyclerViewItemChild(@IdRes int recyclerViewId, int position, @IdRes int itemToClickId) {
-    onView(withId(recyclerViewId)).perform(
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(
         RecyclerViewActions.actionOnItemAtPosition(position, clickChildWithId(itemToClickId)));
   }
 
   public static void clickRecyclerViewItemChild(@IdRes int recyclerViewId, int position, String text) {
-    onView(withId(recyclerViewId)).perform(
+    onView(allOf(withId(recyclerViewId), isDisplayed())).perform(
```

Comment:
Thanks @Sloy ! Just done it on #51 , that applies the same rule over all Barista Actions. By the way, right now, I don't think that this could be a good idea on Assertions, cos we're trying to be aggressive on the assertion part of the test. We can be happy on the when, but not on the end, if we want to make reliable tests.

But I can be wrong. Let's try that approach and fix it if it doesn't work! :tada: